### PR TITLE
controlnet sd 2.1 checkpoint conversions

### PR DESCRIPTION
**Pull request by williamberman**
re: https://github.com/huggingface/diffusers/issues/2581

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py [hunk 1]
```diff
@@ -1043,7 +1062,9 @@ def load_pipeline_from_original_stable_diffusion_ckpt(
         print("global_step key not found in model")
         global_step = None
 
-    if "state_dict" in checkpoint:
+    # NOTE: this while loop isn't great but this controlnet checkpoint has one additional
+    # "state_dict" key https://huggingface.co/thibaud/controlnet-canny-sd21
+    while "state_dict" in checkpoint:
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py:1067]: ok for me

**Review by patrickvonplaten — APPROVED** [review]
Cool looks nice!

**Comment by takuma104** [comment]
@williamberman Thanks for opening the PR! Regarding the URL of `original_config_file`, how about using the URL of `lllyasviel/ControlNet`? I think this file will be used in most use cases.

- For SD1.5: https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v15.yaml
- For SD2.1: https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v21.yaml

**Comment by Piscabo** [comment]
How can I use this in the current Contolnet SD Automatic1111 setup? Please...

**Comment by williamberman** [comment]
> @williamberman Thanks for opening the PR! Regarding the URL of `original_config_file`, how about using the URL of `lllyasviel/ControlNet`? I think this file will be used in most use cases.
> 
> * For SD1.5: https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v15.yaml
> * For SD2.1: https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v21.yaml

Good point! I'm just going to make the config file mandatory as we have no global step to infer the version off of

**Pull request merged**